airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -800,6 +800,10 @@ definitions:
         title: Whether the target API is formatted as a data feed
         description: A data feed API is an API that does not allow filtering and paginates the content from the most recent to the least recent. Given this, the CDK needs to know when to stop paginating and this field will generate a stop condition for pagination.
         type: boolean
+      is_client_side_incremental:
```

Comment:
@artem1205 could we automatically set this to true if `end_datetime` and `start_datetime` are `None`?

rationale is: if we're not configuring the requests sent to the API, then we need to do the filtering on the client side.

the benefit is it would be simpler to configure for the developer since it would be enabled magically

Reply:
Hmm, I think yes, it can be implemented.

Reply:
Added.
I _thought_ about and implemented it with a dependency on `start_time_option`.
But the creation logic became a mess, and maybe we should consider moving our constraints to the Pydantic validator models themselves.

Reply:
IMHO, I prefer to leave the `is_client_side_incremental` property set **explicitly** to be sure what type is used.

Reply:
changes were reverted.
follow up PR to enable automatically: https://github.com/airbytehq/airbyte/pull/38856